src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -370,6 +371,23 @@ public Response getDataset(@PathParam("datasetId") Integer datasetId) {
     }
   }
 
+  @GET
+  @Path("/study/{studyId}")
+  @Consumes("application/json")
+  @Produces("application/json")
+  @RolesAllowed({ADMIN, CHAIRPERSON, DATASUBMITTER})
+  public Response getStudyById(@PathParam("studyId") Integer studyId) {
+    try {
+      Study study = datasetService.findStudyById(studyId);
```

Comment:
This really needs to return more than just the study info. The caller will need to have all of the associated datasets in addition to the study. Currently, this just returns a study with a list of dataset ids, but we should additionally pull the datasets as well.

Reply:
This still returns just a basic study object - it doesn't contain the datasets the UI will need. There are a couple ways to do this, the preferred approach would be to generate a new sql query that pulls all data together. Another approach is to first grab the study in the service class, then query on the dataset ids in a second call. Either one will accomplish the basic task but option 1 is more complex since we don't have that code now while we have all the code in place for option 2.